docs/blueprints/001-installation-of-extensions.md

```diff
@@ -0,0 +1,128 @@
+
+## Proposed solution
+
+### Makefile
+
+Get rid of specific installation targets that install something for production usage (e.g. not used by testing/development).
+
+### CLI
+
+1. Implement new option for `install` command
+
+  `fuseml-installer install [--extension <name>] [--extension-repository <path>]`
+
+  This command installs all necessary components AND extensions selected by `--extension` option. Optionally uses path specified
+  by `--extension-repository` option as a source for extensions that should be installed. Path could be local directory on URL.
+
+
+2. Implement new command for managing extensions. This could be useful for adding an extension to already configured system.
+
+  `fuseml-installer extension`
+
+  This should have several sub commands, like
+
+  * `fuseml-installer extension list` - list installed extensions
+  * `fuseml-installer extension available [--extension-repository <path>]` = list extensions available for installation
+  * `fuseml-installer extension add <name> [--extension-repository <path>]` - install selected extension
+  * `fuseml-installer extension remove <name> [--extension-repository <path>]` - ubinstalls selected extension
+
+3. Add information about extensions into the fuseml-installer config file
+
+### Description files
+
+Extension repository is a directory (local or remote) containing description files of extensions. Each description file provides 
+description of operations that could be done with the extension.
+
+Each extension has its own subdirectory that matches the extension name. Mandatory component under each extension subdirectory is the
+description file, in `yaml` format.
+
+Directory structure:
+
+```
+extensions
+ - installation
+    - mlflow
+      - description.yaml
+      - values.yaml
+    - knative
+      - description.yaml
+      - install.sh
+      - uninstall.sh
+```
+
+Possible installation types for extensions:
+
+* `helm` - helm chart. For helm chart type, no extra install/uninstall commands are necessary
+* `manifest` - Kubernetes manifest, to be installaded using `kubectl`. No extra install/uninstall commands are necessary. All information (including namespace for example) are expected to be present in the manifest file.
+* `script` - shell script. Specific `install` and `uninstall` actions need to be provided by way of referencing specific scripts.
+
+`script` type seems like a least secure one, and we should aim for replacing it with the other types in the future.
+
+
+#### Examples of extension files
+
+```yaml
+name: mlflow 
+description: |
+  MLFlow is an open source platform specialized in tracking ML experiments, and packaging and deploying ML models.
+namespace: 
+install:
+  - type: helm
+    url: https://github.com/fuseml/extensions/charts/mlflow
+    values: https://github.com/fuseml/extensions/installation/mlflow/values.yaml
+  - type: manifest
+    url: https://github.com/fuseml/extensions/installation/mlflow/post-install.yaml
```

Comment:
There seems to be too much duplication needed concerning URLs, especially if you have everything in the same location as the `description.yaml` file. Could the `url` field point to a local path/file that is relative to the `description.yaml` file (or can be absolute, whatever) ? if yes, then please add an example to cover that scenario, something like this:

```
install:
  - type: helm
    url: https://github.com/fuseml/extensions/charts/mlflow
    values: https://github.com/fuseml/extensions/installation/mlflow/values.yaml
  - type: manifest
    url: post-install.yaml
```

Of course, now calling it `url` doesn't work anymore, you'll need another name for it.

Reply:
`location`